main.tf

```diff
@@ -0,0 +1,377 @@
+  sku               = "Standard"
+  sku_tier          = "Regional"
+}
+
+resource "azurerm_backup_protected_vm" "this" {
+  count = var.enable_backup_protected_vm ? 1 : 0
+
+  resource_group_name = local.backup_resource_group_name
+
+  backup_policy_id    = var.backup_policy_id
+  recovery_vault_name = local.backup_recovery_vault_name
+  source_vm_id        = local.virtual_machine.id
+}
+
+resource "random_password" "this" {
+  count  = local.create_password ? 1 : 0
+  length = 24
+}
+
+resource "tls_private_key" "this" {
+  count     = local.create_ssh_key_pair ? 1 : 0
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+#trivy:ignore:avd-azu-0017
+#trivy:ignore:avd-azu-0013
+resource "azurerm_key_vault_secret" "this" {
+  count = var.store_secret_in_key_vault ? 1 : 0
+
+  name         = "${var.name}-${var.admin_username}-${lower(var.authentication_type)}"
+  content_type = var.authentication_type
+  key_vault_id = var.key_vault_id
+  value        = coalesce(local.admin_password, local.admin_ssh_private_key)
+}
+
+resource "azurerm_user_assigned_identity" "this" {
+  count = local.create_identity ? 1 : 0
+
+  name                = "id-${trimprefix(var.name, "vm-")}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+}
+
+resource "azurerm_managed_disk" "this" {
+  for_each = { for element in local.data_disks : element.name => element }
+
+  name                = each.value.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+
+  storage_account_type = each.value.storage_account_type
+  create_option        = each.value.create_option
+  disk_size_gb         = each.value.disk_size_gb
+}
+
+resource "azurerm_virtual_machine_data_disk_attachment" "this" {
+  for_each = { for element in local.data_disks : element.name => element }
+
+  managed_disk_id    = azurerm_managed_disk.this[each.key].id
+  virtual_machine_id = local.virtual_machine.id
+  lun                = each.value.lun
+  caching            = each.value.caching
+}
+
+resource "azurerm_virtual_machine_extension" "this" {
+
+  for_each = { for element in [
+    {
+      name                       = "NetworkWatcherAgent"
+      publisher                  = "Microsoft.Azure.NetworkWatcher"
+      type                       = local.is_windows ? "NetworkWatcherAgentWindows" : (local.is_linux ? "NetworkWatcherAgentLinux" : null)
+      type_handler_version       = "1.0"
+      auto_upgrade_minor_version = true
+      automatic_upgrade_enabled  = true
+    },
+    {
+      name                       = "AzureMonitorAgent"
+      publisher                  = "Microsoft.Azure.Monitor"
+      type                       = local.is_windows ? "AzureMonitorWindowsAgent" : (local.is_linux ? "AzureMonitorLinuxAgent" : null)
+      type_handler_version       = "1.0"
+      auto_upgrade_minor_version = true
+      automatic_upgrade_enabled  = true
```

Comment:
From my point of view, it should be discussed, if the AzureMonitorAgent should be rolled out that way.

Background: AzureMonitor required an data collection role and a managed identity. A system managed identity can be used for that. However some scenarios (like A4L) are using a user-assigned identity. I know that vSOC / Sentinel is used that Agent to.

In general, Microsoft also provide ready policies for that scenario: https://learn.microsoft.com/en-us/azure/azure-monitor/agents/azure-monitor-agent-policy

Reply:
We will discuss this with the Azure chapter, too.

Reply:
At least for this point, I had a discussion with @Wojciech-Rozanski-CDT

Reply:
Will be covered in issue #6.